ui/index.d.ts

```diff
@@ -10,6 +10,7 @@ import { Uri, TreeDataProvider, Disposable, TreeItem, Event, OutputChannel, Meme
 import TelemetryReporter from 'vscode-extension-telemetry';
 import { ResourceGroup } from 'azure-arm-resource/lib/resource/models';
 import { StorageAccount, CheckNameAvailabilityResult } from 'azure-arm-storage/lib/models';
+import { TestAzureAccount } from './src';
```

Comment:
Don't import TestAzureAccount like this. You actually have to add the typing to this file so that it's accessible outside the package.

When you add the typing, please add documentation on the exact environment variables that need to be set for it to work

Reply:
I've discovered I don't like using npm link to test packages - it won't catch things like this until you publish and discover it doesn't actually work.

I prefer to use 'npm pack' and then use 'npm install <path-to-tgz>' which should test exactly the same as if it were published.

Reply:
Not sure what you mean by "it won't catch things like this". I linked his package as-is, added a reference to TestAzureAccount in a random file in functions, and now I see this build error:
![screen shot 2018-08-09 at 10 32 59 am](https://user-images.githubusercontent.com/11282622/43915515-d03dfc8c-9bbf-11e8-8741-b516b4163a23.png)

All I had to do was add `export declare class TestAzureAccount {}` to index.d.ts and remove `import { TestAzureAccount } from './src';` and now the error is gone:
![screen shot 2018-08-09 at 10 32 14 am](https://user-images.githubusercontent.com/11282622/43915509-c9586ae2-9bbf-11e8-982e-b7f6ed830793.png)

I will admit that the code will run if he ignores build errors, but we shouldn't be ignoring build errors :P